### PR TITLE
uhdm-yosys: fix testing ibex-core

### PR DESCRIPTION
**Pull request by kamilrakoczy**
This PR fixes building ibex-core using uhdm-yosys.

Signed-off-by: Kamil Rakoczy <krakoczy@antmicro.com>

### CODE REVIEW

**Comment by hzeller** [comment]
Since this is refining fusoc to create the file list, would this also fix #1400 and #1347 ?

**Comment by tgorochowik** [comment]
@hzeller since fusesoc generates config meant for specific tools (verilator and yosys in this case) the best thing we can do is adding a filter to make sure the tests only run on those.

That would mean skipping other tools completely. I am not sure if there is a way to generate a generic/tool-agnostic config with fusesoc.
If there is, then this would be a great next step, for now we intend to ignore most of the tools. So you could say those issues will be fixed _in a way_.

File: generators/fusesoc [hunk 1]
```diff
@@ -29,11 +29,26 @@ except IndexError:
...
     sys.exit(1)
 
+try:
+    runner = os.environ['RUNNERS_FILTER']
+except KeyError:
+    runner = None
+
+tests_list_exclude = []
+
+if runner in ["UhdmYosys"]:
```
Comment (tgorochowik) [inline generators/fusesoc:39]: This needs to be moved to the generic runner code (as discussed offline).

The runners need to check whether a specific test is intended for them and then skip it if it's not. This implementation will not work

**Comment by hzeller** [comment]
In this particular case, most tools seem to just be stumbling upon the *.vlt file. Can we hack around that and just filter that out for other tools ?

File: generators/fusesoc (hunk 1 above)
Comment (kamilrakoczy) [reply generators/fusesoc]: Done, I added new meta-tag: ``compatible-runners`` to check whether a specific test is for given runner.

**Comment by kamilrakoczy** [comment]
> In this particular case, most tools seem to just be stumbling upon the *.vlt file. Can we hack around that and just filter that out for other tools ?

I've added this hack and now ``*.vlt`` files (for ``ibex`` test) should only go to verilator tools.

**Review by tgorochowik — APPROVED** [review]

**Pull request merged**